docs/logging-requests.md

```diff
@@ -0,0 +1,178 @@
+# Logging requests in the Microsoft Graph .NET Client Library
+
+The SDK does not log of request/response infomation out of the box. If you wish to log request and response information performed by the sdk, there are currently two ways do this. These are :
+
+    a. Implement a LoggingHandler and add it to the list of handlers.
+    b. Take advantage of OpenTelemetry.NET's instrumentation of HttpClient
+
+## a. Implementing a LoggingHandler
+
+### Step 1: Implement a HttpClient Message Handler
+
+You can read more about HttpClient Message Handlers [here](https://docs.microsoft.com/en-us/aspnet/web-api/overview/advanced/httpclient-message-handlers).
+
+The implemented logging handler will look something like the example below. The example simply logs a statement to show that the request has been sent out and the response has come back.
+
+```cs
+using Microsoft.Extensions.Logging;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GraphResponseSample
+{
+    public class LoggingHandler : DelegatingHandler
+    {
+        private readonly ILogger<LoggingHandler> _logger;
+
+        public LoggingHandler(ILogger<LoggingHandler> logger) {
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Sends a HTTP request.
+        /// </summary>
+        /// <param name="httpRequest">The <see cref="HttpRequestMessage"/> to be sent.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request.</param>
+        /// <returns></returns>
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage httpRequest, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Sending Graph request", httpRequest);// log the request before it goes out.
+            HttpResponseMessage response = await base.SendAsync(httpRequest, cancellationToken);
+            _logger.LogInformation("Received Graph response", response);// log the response as it comes back.
+            return response;
+        }
+    }
+}
+```
+
+### Step 2: Add the Handler to GraphServiceClient's list of handlers
+
+Once the logging handler is implemented. You can add it to the list of handlers used by the SDK as follows.
```

Comment:
We should specify the best practice of adding it at the end of the handler list so that it captures all requests/responses regardless of what the earlier handlers added.

Reply:
This is valid. Will address this in a separate PR.

Reply:
Resolved through #235